docs/contributing/supporting_new_profiler.md

```diff
@@ -0,0 +1,109 @@
+# Supporting New Taxonomic Profilers
+
+A good way to contribute to the taxpasta project, is to add support for a new taxonomic profiler. This mostly boils down to creating three new Python modules and filling them with life.
+
+The taxpasta package is designed to follow a [hexagonal architectural](https://en.wikipedia.org/wiki/Hexagonal_architecture_(software)) style. That means, the directory structure may be somewhat unfamiliar to you. Fortunately, three new Python modules only need to be placed in `taxpasta/infrastructure/application/` so it's rather simple.
+
+What do those three new modules need to do? Basically, taxpasta needs to be able to read a taxonomic profile, validate its correctness as much as possible, and finally transform it to what we call the [`StandardProfile`][taxpasta.domain.standard_profile.StandardProfile]. All further processing and logic is based on the `StandardProfile` so you don't need to change anything else!
+
+```mermaid
+graph LR
+  A[Taxonomic Profile] -->|read| B[pandas.DataFrame];
+  B -->|validate| C{ };
+  C -->|transform| D[StandardProfile];
+  C --> E[Exception];
+```
+
+## 1. Read
+
+For reading a taxonomic profile from a new tool, you need to inherit a new class from the abstract [`ProfileReader`][taxpasta.application.ProfileReader]. This new class should have a name corresponding to the tool that you want to support, as an example `Kraken2ProfileReader` to support `kraken2`. Your new module should be named like your class, except all lowercase and underscores `_` separating words, e.g., `kraken2_profile_reader.py`. Since you need to import your base class from a different package branch, you should use an absolute import.
+
+```python
+from taxpasta.application import ProfileReader
+
+
+class Kraken2ProfileReader(ProfileReader):
+```
+
+## 2. Validate
+
+We are assuming that all profile data is tabular and therefore [pandas](https://pandas.pydata.org/) is our choice for handling this data. In order to perform validation of tabular data, we use [pandera schema models](https://pandera.readthedocs.io/en/stable/schema_models.html). The validation is automatically performed by decorating and annotating the `transform` classmethod of the standardisation service, for example,
```

Comment:
Mention if not tabular, ask on the slack channel?

Reply:
Will add that 👍🏼